validator/likecoin-chain-node/setup-a-node.md

```diff
@@ -1,108 +1,440 @@
+
+By default, binaries will be downloaded to the user's home directory `~/`, and the liked data directory will be `~/.liked`
+
+For advanced configuration, such as changing folder locations or specifying binary versions, refer to [the developer facing guide](https://github.com/likecoin/likecoin-chain/blob/master/SETUP.md)
+
+> If you are migrating an existing node on the same host, please ensure that your existing `.liked` folder is not at your home directory, otherwise this might overwrite your files!
+
+For latest genesis url and seed node, refer to:
+
+- [Mainnet](https://github.com/likecoin/mainnet/)
+- [Public Testnet](https://github.com/likecoin/testnets)
+
+```shell
+cd ~/likecoin-chain
+export MONIKER='My Validator'
+export GENESIS_URL='https://raw.githubusercontent.com/likecoin/mainnet/master/genesis.json'
+export LIKED_SEED_NODES='913bd0f4bea4ef512ffba39ab90eae84c1420862@34.82.131.35:26656,e44a2165ac573f84151671b092aa4936ac305e2a@nnkken.dev:26656'
+make -C deploy/cmd/deploy/assets setup-node
+```
+
+### Import existing validator files
+
+> For migrating existing validators only. New operators should skip this section.
+
+First, rename the `.liked` folder generated by the script as `.liked.template`:
+
+```shell
+mv ~/.liked ~/.liked.template
+```
+
+> Below assumes the old docker guide was used. Please adopt suitable steps to stop the existing node and move `.liked` for customized setups.
+
+Then, stop your existing node:
+
+```shell
+cd /path/to/old/repo/likecoin-chain
+docker-compose down
+```
+
+Move the old `.liked` folder to `~/.liked`:
+
+```shell
+cd /path/to/old/repo/likecoin-chain
+mv ./.liked ~/.liked
+```
+
+Add the new `cosmovisor` sub-folder created by the script to `~/.liked`:
+
+```shell
+cp -r ~/.liked.template/cosmovisor ~/.liked/cosmovisor
+```
+
+Check that there are 4 folders within `~/.liked`:
+
+```shell
+config
+cosmovisor
+data
+keyring-file
+```
+
+For some setups, the keyring files may be on the root of `.liked` directly instead of inside `keyring-file` folder, please move the following files to `~/.liked/keyring-file` folder so the liked flag `--keyring-backend=file` works:
+
+```shell
+<hash>.address
+keyhash
+<account name>.info
 ```
+
+### State sync
+
+> Skip this section for migrating operators
+
+Syncing a node from genesis often requires hours if not days. To save time, we can start the new node from a known block.
+
+Firstly, obtain two trusted rpc endpoints. These info can often be found on the same repos hosting genesis files. We have included the addresses of mainnet public nodes below.
+
+Then, we will need to obtain the latest block height and hash from one of the rpc endpoints:
+
+```shell
+curl -s https://fotan-node-1.like.co:443/rpc/block | jq -r '"height: \(.result.block.header.height)\nhash: \(.result.block_id.hash)"'
+```
+
+Next, we will edit `~/.liked/config/config.toml`. Locate the `[statesync]` section and update it to the following values:
```

Comment:
I think we should provide a default config.toml for download.

Reply:
note `config.toml` is for using state sync, while `app.toml` is for serving state sync snapshots (tracking by https://zube.io/oursky/likecoin-chain/c/115)